### PR TITLE
Maxim timer driver

### PR DESCRIPTION
**Pull request by CiprianRegus**
To be merged after  #1377.

### CODE REVIEW

File: drivers/platform/maxim/maxim_irq.c [hunk 1]
```diff
@@ -86,16 +100,21 @@ static bool is_gpio_irq_id(uint32_t irq_id)
...
 }
 
-/**
- * @brief Action comparator function
- * @param data1 - List element
- * @param data2 - Key
- * @return 0 if the two are equal, any other integer otherwise
- */
-int32_t irq_action_cmp(void *data1, void *data2)
+static void _timer_common_callback(mxc_tmr_regs_t *tmr)
```
Comment (amiclaus) [inline drivers/platform/maxim/maxim_irq.c:103]: header comment would be nice.

File: drivers/platform/maxim/maxim_timer.c [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+		MXC_TMR_Start(MXC_TMR_GET_TMR(desc->id));
+
+	desc->freq_hz = freq_hz;
+
+	return 0;
+}
+
+/**
+ * @brief Not implemented
+ *
```
Comment (amiclaus) [inline drivers/platform/maxim/maxim_timer.c:292]: nit: remove line

File: drivers/platform/maxim/timer_extra.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Maxim specific timer platform ops.
+ */
+extern const struct no_os_timer_platform_ops max_timer_ops;
+
+#endif
```
Comment (amiclaus) [inline drivers/platform/maxim/timer_extra.h:58]: nit: add `//_TIMER_EXTRA_H` after #endif

**Review by amiclaus — COMMENTED** [review]
looks good. some minor comments on my side.

**Comment by CiprianRegus** [comment]
- Added indexes to `_events` array.
- Implemented requested changes.

**Review by amiclaus — APPROVED** [review]

**Review by dbogdan — APPROVED** [review]

**Pull request merged**